### PR TITLE
Feat-Wishlist

### PR DESCRIPTION
**Pull request by krishanu2911**
Added Wishlist functionality where users can add a particular product to their Wishlist.

https://user-images.githubusercontent.com/75560307/160486977-55acfb96-72ef-4d4a-9b2a-8a890caac570.mp4

### CODE REVIEW

File: src/component/ProductCard.js [hunk 1]
```diff
@@ -25,7 +37,7 @@ function ProductCard( {prod}) {
                 </div>
               </div>
               <div className="footer-card">
-                <div className="icons txt-lg bold-font">
+                <div className="icons txt-lg bold-font" onClick={() => addWishlistHandler(prod)}>
```
Comment (aishwarya-bm) [inline src/component/ProductCard.js:40]: I think you should have the icon behave like a toggle, meaning that add to wishlist will show a filled heart icon & clicking on that again removes it from the wishlist & shows an empty heart icon. So in your addWishlistHandler also, when the item is already present in the wishlist, clicking the filled heart removes it from the wishlist

**Review by aishwarya-bm — APPROVED** [review]
good breakdown of components, clean and readable code, left a suggestion

File: src/component/ProductCard.js (hunk 1 above)
Comment (krishanu2911) [reply src/component/ProductCard.js:40]: Thank you, Aishwarya for the suggestion done applied the Toggle functionality on the button

File: src/pages/WishlistCard.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import React from "react";
+import "../App.css";
+import { useUserDetail } from "../context/userdetail-context";
+function WishlistCard( { item }) {
+  const { userDetail , setUserDetail } = useUserDetail();
+  const { wishlist } = userDetail;
+  const { imgurl ,title ,rating ,price  } = item;
+  function removeWishHandler (prod) {
+      const RevomveWishList = wishlist.filter((item) => item._id !== prod._id );
```
Comment (sadathshariff) [inline src/pages/WishlistCard.js]: A typo here might cause an issue at later point

**Review by sadathshariff — APPROVED** [review]
Looks Good and good naming convention

File: src/pages/WishlistCard.js (hunk 2 above)
Comment (krishanu2911) [reply src/pages/WishlistCard.js]: Thank you so much Sadath for pointing out

**Pull request merged**